examples/boids_qtree/boids.js

```diff
@@ -0,0 +1,203 @@
+// Daniel Shiffman
+// http://natureofcode.com
+// Boid class based on the Nature of Code
+// approach to autonomous agents.
+
+// Ported by Ali Al-Saqban
+
+// Boid class that utilizes qtree data-structure
+
+class Boid {
+
+    constructor() {
+        // Initializing location to be spread out rather than starting from one point
+        // which affects performance.
+        this.vel = createVector(random(-1,1),random(-1,1));
+        this.acc = createVector(0, 0);
+        this.loc = createVector(randomGaussian(width/2,4),randomGaussian(height/2,4));
+
+        // Other attributes and limits
+        this.r = 7
+        this.maxspeed = 3;
+        this.maxforce = 0.05;
+        this.mass = 1;
+
+    }
+
+    run(qtree) {
+        // The NoC approach took in a Boids list prior to this. Switch to qtree data-structure instead
+        this.flock(qtree);
+        this.update();
+        this.render();
+        this.torroidal();
+    }
+
+    update() {
+        //Update location
+        this.vel.add(this.acc);
+
+        //Limit velocity and update location
+        this.vel.limit(this.maxspeed);
+        this.loc.add(this.vel);
+
+        //Reset acceleration
+        this.acc.mult(0)
+    }
+
+    applyForce(force) {
+        // Rearranging F = ma to a = F/m
+        this.acc.add(force)
```

Comment:
mass is assigned in the constructor but not used here. Either update mass here or remove unused variable?

Reply:
I included it at first, then removed it since I had it at 1 and thought it was an unnecessary division. I will include mass back in here. Thanks for the review :)